samples/account-linking/csharp/README.md

```diff
@@ -21,9 +23,103 @@ The code generically handles account linking for OAuth2.0, the only GitHub speci
 Please see the [Code Tours](#code-tours) section for in-depth explanation of the sample. 
 
 ## About the sample
-    After the user installs the app, they will be prompted when using any of the capabilities (tab / conversational bot / messaging extension) to log into GitHub. After logging in the app will store the users access and refresh tokens so that they won't need to log in again across the other capabilities or on other devices. 
-
-    There are example in-memory and Azure implementations of the token persistance. 
+After the user installs the app, they will be prompted when using any of the capabilities (tab / conversational bot / messaging extension) to log into GitHub. After logging in the app will store the users access and refresh tokens so that they won't need to log in again across the other capabilities or on other devices. 
+
+There are both in-memory and Azure example implementations of the token persistence. 
+    
+You will find the implementation details below. 
+
+This sample will help you
+* Show you the architecture to build out more seamless authentication by using SSO in Teams
+* Use this code (if you build in C#) 
+
+### Scope
+SSO is a problem across multiple surface areas and this sample can be used to build solutions in three surface areas:
+* Tabs
+* Bots
+* Messaging extensions
+
+This sample does not cover:
+* Microsoft Graph tokens. These tokens are received by exchanging user access tokens on the server-side.
+* All languages, it is only built in C#
+* Specific auth systems on your (partner) side
+
+### Context
+#### Problem Overview
+AAD SSO is important for three parties:
```

Comment:
nit: but I feel like either going with something like "AAD SSO provides different benefits to different parties:" reads better than "important for"